### DISCUSSION TITLE
When can we start testing the master branch for the upcoming 4.0.2.4 ?

### DISCUSSION THREAD

**Discussion by mhcwebdesign** [opencart/opencart#13071] [General]
Just asking, because at the moment there are tens, or sometimes hundreds, of daily pull requests, most of them of just a cosmetic nature. Also, there is no point in testing the new OCmod, or using it in extensions, until at least all these crazy daily cosmetic source code changes are done. Any estimates as to when the latter will be the case?

**Comment by julioccguimaraes** [comment]
OpenCart is losing popularity because the project doesn't focus on fixing bugs. The project should stop introducing new features for a while until it releases a stable version of OpenCart 4 that people won't be afraid to install in a production environment.

**Reply by decMuc** [reply]
I think, the way things are progressing at the moment, Daniel is on a good way. The project is so extensive that I have the utmost respect for it.

However, I also see a significant advantage of OCMOD here.

I'm just waiting for OCMOD to be implemented and fully functional again. If I find a bug and figure out what's causing it, I can report the bug, potentially override it temporarily with OCMOD, and once the issue is fixed, remove the temporary modification.

This way, I don't have to wait for Daniel to address all the bugs; I can start right away and possibly contribute to speeding up the completion of OC4.

**Reply by felipoantonoff** [reply]
I agree, I commented on the launch of Opencart 4 that it was very early and indirectly affected Opencart 3, as new stores want to go with the most current version, but the most current version is not yet recommended for production.

In any case, I'm happy to see the return of OCMOD, which I've been commenting on for over a year as Opencart 4's biggest mistake, as it's the best development system for Extensions and one of Opencart's biggest differentiators.

I really agree with the focus on corrections, Opencart 4 had a lot of focus on new features that should be done by Extensions and they affected development time too much, like the new Checkout, it was more important to fix Bugs and create a new theme for later be thinking about more resources, now all we have to do is wait and help as we can until Opencart 4 is finally ready for production and with good extensions made for it, with the return of OCMOD it will greatly strengthen Opencart again and many developers who were away from Opencart 4 they can start focusing on it.

**Comment by IP-CAM** [comment]
Well, OcMod is now already implemented in the latest Github Version, as is looks.

**Comment by danielkerr** [comment]
no, i need to finish the blog comments section then subscription emails possibly. i need to test them all.

**Reply by felipoantonoff** [reply]
Very good Daniel, I was happy with the Blog system, it would be good to support Markdown in the future, I'll add it later if I'm not using OCMOD.

Now I think it would be good in the future for Opencart to follow a more MVP line, do as little as possible for new features and over time add new features, the Blog itself could have started as just a simple Blog, then Categories, Comments and so on.

**Comment by Hkr32** [comment]
> Just asking, because at the moment there are tens, or sometimes hundreds, of daily pull requests, most of them of just a cosmetic nature. Also, there is no point in testing the new OCmod, or using it in extensions, until at least all these crazy daily cosmetic source code changes are done. Any estimates as to when the latter will be the case?

This cosmetics has been going on since the moment [TheCartpenter](https://github.com/TheCartpenter) appeared, he does what IDE itself or Prettier has been able to do for a long time. Why is he doing this? It's like work for work's sake...

**Comment by danielkerr** [comment]
i think its taking to long now. i will start finishing off features and issues.

**Comment by mhcwebdesign** [comment]
@danielkerr : Once done, can you please first do a public 4.0.2.4.beta release, to have a greater number of testers and users try it out for a few weeks, before doing a final 4.0.2.4 release?
